bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/AbstractExpression.java

```diff
@@ -207,22 +207,51 @@ protected void prune() {
 
     @Override
     public Date getTimeAfter(Date afterTime) {
+
+        Date currentStartDate = getStartDate();
+
+        if (hasFloatingStartDate()) {
+            try {
+                clearCandidates();
+                setStartDate(afterTime);
+            } catch (IllegalArgumentException | ParseException e) {
+                logger.error("An exception occurred while setting the start date : '{}'", e.getMessage());
+            }
+        }
+
         if (getCandidates().isEmpty()) {
```

Comment:
@kgoderis Do you consider to change the `if` to `else if`? I think the remaining stuff is fine and we could merge this PR.